src/Vehicle/Vehicle.cc

```diff
@@ -1492,7 +1492,16 @@ void Vehicle::_updatePriorityLink(void)
     }
 #endif
 
-    if (!newPriorityLink && !_priorityLink.data() && _links.count()) {
+    bool found_priority_link = false;
+    if (!newPriorityLink && _priorityLink.data()) {
+        for (int i=0; i<_links.count(); i++) {
+            if (_priorityLink->getLinkConfiguration() == _links[i]->getLinkConfiguration()) {
```

Comment:
Is there a reason this is not ```if (_priorityLink.data() != link)```?

Reply:
Reason I ask is that in looking at the code for this method it's really confusing. Working on rewriting with your fix.